lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -229,23 +263,33 @@ def set_segments(self, segments):
         """
         Set 3D segments.
         """
-        self._segments3d = np.asanyarray(segments)
+        self._segments3d = []
+        if len(segments) > 0:
+            self._seg_sizes = np.array([len(c) for c in segments])
+            # Store the points in a single array for easier projection
+            n_segments = np.sum(self._seg_sizes)
+
+            self._segments3d_data = np.empty((n_segments, 4))
+            self._segments3d_data[:, :3] = np.vstack(segments)
+            self._segments3d_data[:, 3] = 1
+
+            # For convenience, store a view of the array in the original shape
+            self._segments3d = _array_split(self._segments3d_data[:, :3],
+                                            np.cumsum(self._seg_sizes))
+
         LineCollection.set_segments(self, [])
 
     def do_3d_projection(self, renderer):
         """
         Project the points according to renderer matrix.
         """
-        xyslist = [
-            proj3d.proj_trans_points(points, renderer.M) for points in
-            self._segments3d]
-        segments_2d = [np.column_stack([xs, ys]) for xs, ys, zs in xyslist]
+        if len(self._segments3d) == 0:
+            return 1e9
```

Comment:
Maybe the old fixme should stay here, with a comment explaining its for back-compat. I'd be happier with a change to `float('inf')` here in some future patch.